### ISSUE TITLE
Expose `endpoint` argument in `get_s3fs_session` top-level function

### ISSUE THREAD

**Issue by chuckwondo** [nsidc/earthaccess#602]
Per https://github.com/nsidc/earthaccess/discussions/482, there is a desire to be able to directly invoke `get_s3fs_session` from the top of `earthaccess`, like so:

```python
earthaccess.get_s3fs_session(endpoint="https://nisar.asf.earthdatacloud.nasa.gov/s3credentials")
```

Currently, it is possible to do the following, but accessing `__store__` is perhaps not desirable:

```python
earthaccess.__store__.get_s3fs_session(endpoint="https://nisar.asf.earthdatacloud.nasa.gov/s3credentials")
```

**Comment by jhkennedy** [comment]
yes, this is a good idea. In the same vein, I also find this method useful and think it would be good expose directly:
```python
>>> earthaccess.__store__.auth.get_s3_credentials(endpoint='https://nisar.asf.earthdatacloud.nasa.gov/s3credentials')
{'accessKeyId': 'XXX',
 'secretAccessKey': 'XXX',
 'sessionToken': 'XXX',
 'expiration': '2024-06-12 20:48:21+00:00'}
```
and might be a nice addition to the TEA s3 credentials READMEs: <https://nisar.asf.earthdatacloud.nasa.gov/s3credentialsREADME>

Notably, this does not work:
```python
>>> earthaccess.auth.get_s3_credentials(endpoint='https://nisar.asf.earthdatacloud.nasa.gov/s3credentials')
AttributeError: module 'earthaccess.auth' has no attribute 'get_s3_credentials'
```

**Comment by JessicaS11** [comment]
For anyone interested in working on this issue, there is currently an [`earthaccess.get_s3_credentials()` function](https://github.com/nsidc/earthaccess/blob/7a79ea5f1eb752d9050f58e3e0c2813a4409626d/earthaccess/api.py#L328) exposed at the top level (`api.py`). My understanding is this issue aims specifically to include exposure of the `endpoint` kwarg there.

**Comment by chuckwondo** [comment]
> For anyone interested in working on this issue, there is currently an [`earthaccess.get_s3fs_session()` function](https://github.com/nsidc/earthaccess/blob/7a79ea5f1eb752d9050f58e3e0c2813a4409626d/earthaccess/api.py#L328) exposed at the top level (`api.py`). My understanding is this issue aims specifically to include exposure of the `endpoint` kwarg there.

Yes, sorry if my original description was confusing.

**Comment by mfisher87** [comment]
How's this read to y'all?
